### PR TITLE
feat: removed pkg_resources from all test files and moved importlib into pandas extra

### LINKED ISSUE #313: use native support for namespace packages
Per: https://github.com/googleapis/google-auth-library-python/issues/595#issuecomment-683903062

> It looks like different Google API packages have different approaches to using or not using pkg_resources here; e.g. https://github.com/googleapis/python-cloudbuild uses native NS packages and https://github.com/googleapis/python-bigquery uses pkg_resources with a pkgutil fallback.

Now that Python 2.7 and 3.5 support has been dropped, we may be able to follow the example in python-cloudbuild

### LINKED ISSUE #1725: Remove pkg_resource dependency after native namespace compatibility is implemented
Similar to: https://github.com/googleapis/python-bigquery-storage/pull/706
Pkg_resources can be removed from tests after https://github.com/googleapis/python-bigquery/pull/1720 is merged, importlib can be used instead.

### PR DESCRIPTION
**Pull request by kiraksi**
In this PR
- removed all instances of pkg_resources in tests and introduced importlib instead
- similar to https://github.com/googleapis/python-bigquery-storage/pull/706
- should resolve any further issues we had stemming from pkg_resources

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #1725 🦕
Fixes #313 🦕

### CODE REVIEW

File: tests/system/test_pandas.py
Comment (parthea) [inline tests/system/test_pandas.py:53]:
```suggestion
```


Instead of 
```
@pytest.mark.skipif(
    PANDAS_INSTALLED_VERSION < PANDAS_INT64_VERSION,
    reason="Only `pandas version >=1.0.0` is supported",
)
```
we could have
```
@pytest.mark.skipif(
    PANDAS_INSTALLED_VERSION[0:2] in ["0."],
    reason="Only `pandas version >=1.0.0` is supported",
)
```

File: tests/unit/test__pandas_helpers.py
Comment (parthea) [inline tests/unit/test__pandas_helpers.py:68]: Previously we didn't have `# pragma: NO COVER`. Is it needed?

Comment (kiraksi) [reply tests/unit/test__pandas_helpers.py:68]: When I ran the kokoro build test it was giving me a failure on coverage in a test file (not this one), so I thought to add the tag everywhere just in case. It may have just been an issue with that test file, so I can remove it here and other places if needed!

**Review by parthea — APPROVED** [review]
LGTM. Added one observation for potentially unnecessary version checks in test files.

File: tests/system/test_pandas.py
Comment (chalmerlowe) [reply tests/system/test_pandas.py:317]: I tend to agree, we probably could simplify some of this. I am not sure that should hold up this PR.

Comment (kiraksi) [reply tests/system/test_pandas.py:317]: I can make this change today as well before I planned on merging

File: tests/system/test_pandas.py
Comment (chalmerlowe) [inline tests/system/test_pandas.py:1045]: This has several elements in the list so it is a bit easier to grok what is happening here.

**Review by chalmerlowe — APPROVED** [review]
LGTM

File: tests/system/test_pandas.py
Comment (parthea) [inline tests/system/test_pandas.py:345]:
```suggestion
    PANDAS_INSTALLED_VERSION[0:2].startswith("0."),
    reason="Only `pandas version >=1.0.0` is supported",
```

**Comment by parthea** [comment]
Good catch @tswast ! I can produce the issue when using an editable install using egg as it appears in the [build log](https://fusion2.corp.google.com/invocations/2be1e8d5-99f1-41aa-97e4-4cabb94bd441/targets/bigframes%2Fpresubmit%2Fe2e/log) 

When I install using an editable install using egg, the import fails
```
pip install -e git+https://github.com/googleapis/python-bigquery.git@1f4ebb1eca4f9380a31172fc8cb2fae125f8c5a2#egg=google_cloud_bigquery --ignore-installed
```

When I install without an editable install, the import succeeds
```
pip install https://github.com/googleapis/python-bigquery/archive/main.zip --ignore-installed
```

I can produce the same issue with google auth. The import of `google.auth` fails when I use
```
pip install -e git+https://github.com/googleapis/google-auth-library-python.git@7ab0fced0008ad9283c955b7dd7e2b7db55ae5e7#egg=google_auth --ignore-installed`
```

but not with
```
pip install https://github.com/googleapis/google-auth-library-python/archive/main.zip
```

The change to move to native namespace packages was [released in July](https://github.com/googleapis/google-auth-library-python/releases/tag/v2.22.0) for `google-auth`. 

We should investigate whether we want to block on this issue before reverting the change as we do want to support python 3.12 without using `pkg_resources`.  

We could also consider the change to move to native namespace packages as a potential breaking change and bump the major version.

**Comment by tswast** [comment]
Thanks for the investigation! I think we can remove the "-e" in the BigFrames tests.

**Pull request merged**
